lib/service-host.js

```diff
@@ -42,10 +44,32 @@ class ServiceHost {
 	getScriptSnapshot() {
 		return ts.ScriptSnapshot.fromString(this.content);
 	}
+}
+
+/**
+ * Language Service Host class for Vue support, that has some extra stuff needed by `@volar/vue-typescript`.
+ *
+ * @typedef {import('@volar/vue-language-core/out/types').LanguageServiceHost} VueLanguageServiceHost
+ * @implements {VueLanguageServiceHost}
+ */
+class VueServiceHost extends ServiceHost {
+	/**
+	 * Create a service host instance for the given file.
+	 *
+	 * @param {string} name path to file
+	 * @param {string} content file content
+	 */
+	constructor(name, content) {
+		super(name, content);
+	}
 
 	getVueCompilationSettings() {
 		return {};
 	}
+
+	getTypeScriptModule() {
+		return require('typescript/lib/tsserverlibrary');
```

Comment:
I changed this from `ts` to this because once I added the `@implements` JS Doc tag, it was complaining that `ts` wasn't compatible. @johnsoncodehk do you think this is fine or should it really be `ts`?

Reply:
volar access `typescript` and `typescript/lib/tsserverlibrary` both, you already load `typescript` in memory, `typescript/lib/tsserverlibrary` is a large script so I think reuse `ts` is better.

You can just use as any or `@ts-ignore` to ignore this error.